### PR TITLE
BACK button, LCD key nav, macOS build

### PR DESCRIPTION
**Pull request by thinkyhead**
- Add code to the LCD keyboard for more responsive navigation.
- Add support for `BTN_BACK` for faster "back" using left arrow.
- Override for `platformio.ini` to specify the dependency check compiler.
- Extend `simulator.py` so macOS can fall back to Mesa if OpenGL is not found.
- General code cleanup, suppress some warnings.

I had some other improvements, but those are now merged into `bugfix-2.0.x`, so just merge from there to acquire them.

### CODE REVIEW

File: Marlin/src/HAL/NATIVE_SIM/sim/hardware/ST7920Device.h [hunk 1]
```diff
@@ -20,24 +20,21 @@ class ST7920Device: public Peripheral {
...
-  ST7920Device(pin_type clk, pin_type mosi, pin_type cs, pin_type beeper, pin_type enc1, pin_type enc2, pin_type enc_but, pin_type kill);
+  ST7920Device(pin_type clk, pin_type mosi, pin_type cs, pin_type beeper, pin_type enc1, pin_type enc2, pin_type enc_but, pin_type back, pin_type kill);
   virtual ~ST7920Device();
   void process_command(Command cmd);
   void update();
   void interrupt(GpioEvent& ev);
   void ui_callback(UiWindow* window);
 
-  static constexpr uint32_t width = 128;
-  static constexpr uint32_t height = 64;
```
Comment (rhapsodyv) [inline Marlin/src/HAL/NATIVE_SIM/sim/hardware/ST7920Device.h:31]: @thinkyhead This code must stay, as we have more than one display now.

**Comment by rhapsodyv** [comment]
You need to update the `DISPLAY_PARAM` too:
```
#define DISPLAY_PARAM LCD_PINS_D4, LCD_PINS_ENABLE, LCD_PINS_RS, BEEPER_PIN, BTN_EN1, BTN_EN2, BTN_ENC, ???BTN_BACK?????, KILL_PIN
```

**Comment by rhapsodyv** [comment]
And add `BTN_BACK` to the reprap full discount in the pins file.

File: Marlin/src/HAL/NATIVE_SIM/sim/visualisation.h [hunk 2]
```diff
@@ -212,7 +212,7 @@ class Visualisation {
   bool input_state[6] = {};
   glm::vec<2, int> mouse_lock_pos;
 
-  std::array<GLfloat, 24 * 10> g_vertex_buffer_data{
+  const std::array<GLfloat, 24 * 10> g_vertex_buffer_data{
```
Comment (rhapsodyv) [inline Marlin/src/HAL/NATIVE_SIM/sim/visualisation.h]: This can't be const! It's changed in the code, because it draw the bed and nozzle.

**Comment by p3p** [comment]
I removed your changes to the format string, its a cross platform issue we will have to work around somehow

**Comment by thinkyhead** [comment]
Ah, yes, those strings! We can always do something similar to `#define S_FMT "%S"` in future.

**Pull request merged**